### ISSUE TITLE
RuntimeError: A pipeline has not yet been optimized. Please call fit() first. 

### ISSUE THREAD

**Issue by janPesl** [EpistasisLab/tpot#901]
This issue is most likely related to #499.

I am fitting classifier:
```
tp = TPOTClassifier(
    generations=50,
    population_size=50,
    cv=4,
    n_jobs=-1,
    random_state=0,
    config_dict=classifier_config_dict,
    verbosity=3,
    use_dask=True,
)
```

Before any optimization is done, I keep running into this issue:

```
---------------------------------------------------------------------------
OptionError                               Traceback (most recent call last)
/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in fit(self, features, target, sample_weight, groups)
    751         return self._fitted_imputer.transform(features)
--> 752 
    753     def _check_dataset(self, features, target):

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/gp_deap.py in eaMuPlusLambda(population, toolbox, mu, lambda_, cxpb, mutpb, ngen, pbar, stats, halloffame, verbose, per_generation_function)
    226     :returns: A tuple of two trees.
--> 227     """
    228     # Define the name of type for any types.

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in _evaluate_individuals(self, population, features, target, sample_weight, groups)

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in <listcomp>(.0)

/anaconda3/envs/python373/lib/python3.7/site-packages/stopit/utils.py in wrapper(*args, **kwargs)
    144                     # ``result`` may not be assigned below in case of timeout
--> 145                     result = func(*args, **kwargs)
    146                 return result

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/gp_deap.py in _wrapped_cross_val_score(sklearn_pipeline, features, target, cv, scoring_function, sample_weight, groups, use_dask)
    421     # build a job for cross_val_score
--> 422     tmp_it = Interruptable_cross_val_score(
    423         estimator=sklearn_pipeline,

/anaconda3/envs/python373/lib/python3.7/site-packages/dask_ml/model_selection/__init__.py in <module>
      5 """
----> 6 from ._hyperband import HyperbandSearchCV
      7 from ._incremental import IncrementalSearchCV

/anaconda3/envs/python373/lib/python3.7/site-packages/dask_ml/model_selection/_hyperband.py in <module>
     10 
---> 11 from ._incremental import BaseIncrementalSearchCV
     12 from ._successive_halving import SuccessiveHalvingSearchCV

/anaconda3/envs/python373/lib/python3.7/site-packages/dask_ml/model_selection/_incremental.py in <module>
     22 
---> 23 from ..utils import check_array
     24 from ..wrappers import ParallelPostFit

/anaconda3/envs/python373/lib/python3.7/site-packages/dask_ml/utils.py in <module>
     11 import dask.array as da
---> 12 import dask.dataframe as dd
     13 import numpy as np

/anaconda3/envs/python373/lib/python3.7/site-packages/dask/dataframe/__init__.py in <module>
      3 try:
----> 4     from .core import (
      5         DataFrame,

/anaconda3/envs/python373/lib/python3.7/site-packages/dask/dataframe/core.py in <module>
     84 
---> 85 pd.set_option("compute.use_numexpr", False)
     86 

/anaconda3/envs/python373/lib/python3.7/site-packages/pandas/core/config.py in __call__(self, *args, **kwds)
    226     def __call__(self, *args, **kwds):
--> 227         return self.__func__(*args, **kwds)
    228 

/anaconda3/envs/python373/lib/python3.7/site-packages/pandas/core/config.py in _set_option(*args, **kwargs)
    133             else:
--> 134                 o.cb(key)
    135 

/anaconda3/envs/python373/lib/python3.7/site-packages/pandas/core/config_init.py in use_numexpr_cb(key)
     43     the default is True
---> 44     Valid values: False,True
     45 """

/anaconda3/envs/python373/lib/python3.7/site-packages/pandas/core/computation/expressions.py in <module>
    167 # turn myself on
--> 168 set_use_numexpr(get_option("compute.use_numexpr"))
    169 

/anaconda3/envs/python373/lib/python3.7/site-packages/pandas/_config/config.py in __call__(self, *args, **kwds)
    232     def __call__(self, *args, **kwds):
--> 233         return self.__func__(*args, **kwds)
    234 

/anaconda3/envs/python373/lib/python3.7/site-packages/pandas/_config/config.py in _get_option(pat, silent)
    101 def _get_option(pat, silent=False):
--> 102     key = _get_single_key(pat, silent)
    103 

/anaconda3/envs/python373/lib/python3.7/site-packages/pandas/_config/config.py in _get_single_key(pat, silent)
     87             _warn_if_deprecated(pat)
---> 88         raise OptionError("No such keys(s): {pat!r}".format(pat=pat))
     89     if len(keys) > 1:

OptionError: "No such keys(s): 'compute.use_numexpr'"

During handling of the above exception, another exception occurred:

RuntimeError                              Traceback (most recent call last)
<ipython-input-24-c5de40c7b1e7> in <module>
----> 1 tp.fit(x_train, y_train)

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in fit(self, features, target, sample_weight, groups)
    782         expr: DEAP individual
    783             The DEAP pipeline to be compiled
--> 784 
    785         Returns
    786         -------

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in fit(self, features, target, sample_weight, groups)
    773                 'the features must be a 2-D array and target labels must be a '
    774                 '1-D array.'
--> 775             )
    776 
    777     def _compile_to_sklearn(self, expr):

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in _update_top_pipeline(self)
    851 
    852         # return fitness scores
--> 853         operator_counts = {}
    854         # 4 lists of DEAP individuals, their sklearn pipelines and their operator counts for parallel computing
    855         eval_individuals_str = []

RuntimeError: A pipeline has not yet been optimized. Please call fit() first.
```

When turning off Dask, I get different error:

```
---------------------------------------------------------------------------
_RemoteTraceback                          Traceback (most recent call last)
_RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/anaconda3/envs/python373/lib/python3.7/site-packages/joblib/externals/loky/process_executor.py", line 418, in _process_worker
    r = call_item()
  File "/anaconda3/envs/python373/lib/python3.7/site-packages/joblib/externals/loky/process_executor.py", line 272, in __call__
    return self.fn(*self.args, **self.kwargs)
  File "/anaconda3/envs/python373/lib/python3.7/site-packages/joblib/_parallel_backends.py", line 567, in __call__
    return self.func(*args, **kwargs)
  File "/anaconda3/envs/python373/lib/python3.7/site-packages/joblib/parallel.py", line 225, in __call__
    for func, args, kwargs in self.items]
  File "/anaconda3/envs/python373/lib/python3.7/site-packages/joblib/parallel.py", line 225, in <listcomp>
    for func, args, kwargs in self.items]
TypeError: _wrapped_cross_val_score() got an unexpected keyword argument 'timeout'
"""

The above exception was the direct cause of the following exception:

TypeError                                 Traceback (most recent call last)
/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in fit(self, features, target, sample_weight, groups)
    751         return self._fitted_imputer.transform(features)
--> 752 
    753     def _check_dataset(self, features, target):

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/gp_deap.py in eaMuPlusLambda(population, toolbox, mu, lambda_, cxpb, mutpb, ngen, pbar, stats, halloffame, verbose, per_generation_function)
    226     :returns: A tuple of two trees.
--> 227     """
    228     # Define the name of type for any types.

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in _evaluate_individuals(self, population, features, target, sample_weight, groups)

/anaconda3/envs/python373/lib/python3.7/site-packages/joblib/parallel.py in __call__(self, iterable)
    933             with self._backend.retrieval_context():
--> 934                 self.retrieve()
    935             # Make sure that we get a last message telling us we are done

/anaconda3/envs/python373/lib/python3.7/site-packages/joblib/parallel.py in retrieve(self)
    832                 if getattr(self._backend, 'supports_timeout', False):
--> 833                     self._output.extend(job.get(timeout=self.timeout))
    834                 else:

/anaconda3/envs/python373/lib/python3.7/site-packages/joblib/_parallel_backends.py in wrap_future_result(future, timeout)
    520         try:
--> 521             return future.result(timeout=timeout)
    522         except LokyTimeoutError:

/anaconda3/envs/python373/lib/python3.7/concurrent/futures/_base.py in result(self, timeout)
    431             elif self._state == FINISHED:
--> 432                 return self.__get_result()
    433             else:

/anaconda3/envs/python373/lib/python3.7/concurrent/futures/_base.py in __get_result(self)
    383         if self._exception:
--> 384             raise self._exception
    385         else:

TypeError: _wrapped_cross_val_score() got an unexpected keyword argument 'timeout'

During handling of the above exception, another exception occurred:

RuntimeError                              Traceback (most recent call last)
<ipython-input-27-c5de40c7b1e7> in <module>
----> 1 tp.fit(x_train, y_train)

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in fit(self, features, target, sample_weight, groups)
    782         expr: DEAP individual
    783             The DEAP pipeline to be compiled
--> 784 
    785         Returns
    786         -------

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in fit(self, features, target, sample_weight, groups)
    773                 'the features must be a 2-D array and target labels must be a '
    774                 '1-D array.'
--> 775             )
    776 
    777     def _compile_to_sklearn(self, expr):

/anaconda3/envs/python373/lib/python3.7/site-packages/tpot/base.py in _update_top_pipeline(self)
    851 
    852         # return fitness scores
--> 853         operator_counts = {}
    854         # 4 lists of DEAP individuals, their sklearn pipelines and their operator counts for parallel computing
    855         eval_individuals_str = []

RuntimeError: A pipeline has not yet been optimized. Please call fit() first.
```

Do you have any suggestions how to get around this? Thank you in advance!

**Comment by weixuanfu** [comment]
I caught this line `_wrapped_cross_val_score() got an unexpected keyword argument 'timeout'` in the stdout, which is weird since the function was wrapped by `
@threading_timeoutable` decorator from `stopit`. Could you please let us know more details about his issue, like OS, versions of TPOT and its dependencies? A demo for reproducing this issue is preferred.

**Comment by janPesl** [comment]
While working on the demo, I have followed this installation instructions https://epistasislab.github.io/tpot/installing/ instead of just `conda install -c conda-forge tpot `.
In the new virtualenv, I couldn't reproduce the issue anymore. When I have tried to follow the same installation procedure again in the previous virtualenv, it has turned into an utter mess - I can't even import pandas anymore. The solution therefore is to scrape the old virtualenv and keep conda and pip strictly separated at all times...
